mimic/core.py

```diff
@@ -128,9 +153,17 @@ def entries_for_tenant(self, tenant_id, prefix_map, base_uri):
         :return: The full URI locating the service for that region
         """
         for service_id, api in self._uuid_to_api.items():
-            for entry in api.catalog_entries(tenant_id):
-                for endpoint in entry.endpoints:
-                    prefix_map[endpoint] = self.uri_for_service(
-                        endpoint.region, service_id, base_uri
-                    )
-                yield entry
+            if IExternalAPIMock.providedBy(api):
+                for entry in api.catalog_entries(tenant_id):
+                    for endpoint in entry.endpoints:
+                        prefix_map[endpoint] = api.uri_for_service(
+                            endpoint.region, service_id
+                        )
+                    yield entry
+            else:  # IAPIMock.providedBy(api)
```

Comment:
I don't think we're iterating the loop twice here. Each entry in `self._uuid_to_api` each object can only be either an Internal API or an External API but not both. If we actually had an `_all_external_mocks` and `_all_internal_mocks` then yes, we would have to iterate over the data twice. However, if we were going to do that then I would split them into separate storage specifically to keep from looping over all the data twice.

I can adjust the comment if you like.